src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -28,6 +29,8 @@ export interface EncodingShelfPropsBase extends ActionHandler<ShelfEncodingActio
   id: ShelfId;
 
   fieldDef: ShelfFieldDef;
+
+  schema?: Schema;
```

Comment:
Because if it's wildcard fields or Number of Record field, they don't need the schema since they cannot have filters.